tests/lent/tbasic_lent_check.nim

```diff
@@ -14,3 +15,31 @@ proc main =
   doAssert(not compiles(passToVar(viewInto(x))))
 
 main()
+
+
+#------------------------------------------------------------------------------
+# issue #15958
+
+block:
+  proc byLent[T](a: T): lent T = a
+  let a = [11,12]
+  let b = @[21,23]
+  doAssert byLent(a) == [11,12]
+  doAssert byLent(a).unsafeAddr == a.unsafeAddr
+  doAssert byLent(b) == @[21,23]
+  doAssert byLent(b).unsafeAddr == b.unsafeAddr
+
+  proc byLent2[T](a: openarray[T]): lent T = a[0]
+  doAssert byLent2(a) == 11
+  doAssert byLent2(a).unsafeAddr == a[0].unsafeAddr
+  doAssert byLent2(b) == 21
+  doAssert byLent2(b).unsafeAddr == b[0].unsafeAddr
+
+
+  proc byLent3[T](a: varargs[T]): lent T = a[1]
+  let 
+    x = 10
+    y = 20
+    z = 30
+  doAssert byLent3(x, y, z) == 20
```

Comment:
also check VM by wrapping everything inside a
```nim
template main =
  ...
static: main()
main()
```

Reply:
I am not fixing VM or JS in this PR

Reply:
=> https://github.com/nim-lang/Nim/pull/15970#discussion_r524396526